### ISSUE TITLE
[FEATURE REQUEST] Classy Neon/Arcade Sona (8bit neon)

### ISSUE THREAD

**Issue by za419** [kenellorando/cadence#95]
As requested by @KarenwithaKay

Either gamer/classy neon to make it look nice, because neon done wrong looks tacky.

"each letter in the Cadence title consecutively flashes neon. the background can contrast with a dark but brightly lit city. or black background for simplicity. i dunno" [for city, I thought of Vegas or New York]

I'll work with her to see if the theme matches her expectations. She is the designer on this one.

**Comment by za419** [comment]
Inspired by [this](https://youtu.be/rAL9gEqpsDs)

**Comment by kenellorando** [comment]
I once had Cyberpunk flash a single letter like a broken sign. 

This code does a similar "broken" style flash animation and is live but not active, so I'd first make an even simpler on/off animation:
https://github.com/kenellorando/cadence/blob/master/public/css/themes/cyberpunk-bartender.css#L39

Then in theme.json I'd span each letter with a unique ID in the new theme object like it is now with star guardian
https://github.com/kenellorando/cadence/blob/master/public/js/theme.json#L49

Then I'd set each span ID to a different time on the animation
https://github.com/kenellorando/cadence/blob/master/public/css/themes/cyberpunk-bartender.css#L28

**Comment by za419** [comment]
That was actually what I was going to steal. Thanks for finding it for me, even though its still on master....?

**Comment by kenellorando** [comment]
Makes it simpler if I ever wanted to "break" a sign letter again

**Comment by za419** [comment]
@kenellorando kennnnnnn

That's literally why we use git..................................

kennnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn

**Comment by kenellorando** [comment]
We *could* remove the code and scroll through 700 commits until we find the ones where it was active...

orrr we could just leave it

**Comment by kenellorando** [comment]
![](https://imgflip.com/s/meme/Roll-Safe-Think-About-It.jpg)

**Comment by za419** [comment]
Speaking of which, why is the ISS warning style code still in the dropdown module....?

**Comment by za419** [comment]
Please tell me you're not keeping it there for the ease of turning it back on

**Comment by kenellorando** [comment]
No, and I remember now exactly why it's there. The `#ISSwarning`, we can remove now. The distinction with the blinker code is that the `#ISSwarning` no longer has potential to be used again. I still fail to see this is a problem. Another related point is the impact on performance / code complexity is negligible to work it would take to re-implement everything again. 

Think of it in a similar way: Just because I don't have an announcement on the page doesn't mean I need to delete all the `h3` and `.announcement` styling, no? When I want to put something back again the styling is still there to support it. When I want to put spans around the letters, the CSS is ready to support animating them.

The animation is a feature we can use if we want, like a theme itself; when a theme is not in rotation, we don't delete it just because git has it saved, we leave it there for potential re-enabling in the future. In the same vein, we don't need to delete a some features just because the element isn't necessarily being used at the time either. So the code will stay

**Comment by za419** [comment]
See #103.

Here's the thing. The point of git, or at least part of it, is that you don't have to re-implement old code, because you have history of it. So the work it would take to re-implement everything again is nominally approaching zero. Of course, you mentioned searching through 600 commits for the feature, for which git provides for you the ability to name a commit, either as a branch or as a lightweight tag, so you don't have to search through any commits for the feature. Remember #89? I didn't dig through old code to find Space Station, I checked out my branch `ex-feature-space-station`, which served as a pointer in case I ever wanted to look at the Space Station code again. Which is to say, Space Station didn't sit in the working directory all this time just in case we wanted to look at it again, it only stayed in git.

So any performance penalty at all, multiplied per time the page is fetched, doubled because dead code incurs one penalty on the server (additional processed payload data), and one penalty on the client (sum of receiving additional data+processing deadweight), severely outweighs the theoretical penalty of running `git show` or `git branch` to find and either merge or copy-paste the relevant data. It's like saying a lobster could become heavier than a blue whale: In theory, it can happen, but in practice, don't expect it. Granted, the difference is tiny... But why leave a penalty in place when you have no reason to?

Keeping `h3` and `.announcement` styling respects the separation of content and style: That is, the CSS shouldn't need to change in normal use of the page. Because these styles are occasionally used, yeah, that's acceptable. Themes stay in service both because `themeChanger` will break if someone is using an out-of-rotation theme, and because we both know that most of our users bypass the rotation anyhow. It's a feature. We're leaving on page content as is - And besides, a theme is not downloaded or processed until a user chooses it. Loading an animation, however, forces the browser to form an in-memory representation of how to play that animation, meaning that it must download a larger file, then spend more time parsing it, and use more memory storing it...... All so that we avoid all of two git commands? I fail to see how this is an even trade.

Cyberpunk's blinking letter isn't switched around on a day-to-day basis. Users don't enable it at their leisure, and it's not a part of adding content to enable it. It's a change to the theme itself, which happens pretty rarely. And, a big reason to use git is that we don't save it there, because we can find it in the past. Your argument can be equally well applied to any number of lines of code that we've removed because new code is better, and we don't keep it around for future reference. git is there for that.
